lua/starfall/libs_sv/cfc_ply.lua

```
@@ -328,6 +328,56 @@ function player_methods:stripWeapon( class )
     ply:StripWeapon( class )
 end
 
+--- Applies force to the player.
+-- @param vector Force
+function player_methods:applyForce( force )
+    checktype( self, ply_meta )
+
+    local ply = getPly( self )
+    checkPlyCorePerms( instance, ply, "applyForce", force )
+
+    ply:SetVelocity( vunwrap( force ) )
+end
+
+--- Sets the mass of the player. Default 85
+-- @param number Mass
+function player_methods:setMass( mass )
+    checkluatype( mass, TYPE_NUMBER )
+    checktype( self, ply_meta )
+
+    local ply = getPly( self )
+    checkPlyCorePerms( instance, ply, "setMass", mass )
+
+    ply:GetPhysicsObject():SetMass( math.Clamp( mass, 1, 50000 ) )
```

Do you need to check physobj validity first?

---

I believe the function should error if the physics object isn't valid because it wouldn't make sense for the function to be successful if it doesn't have a valid physics object.

---

Would the error be useful to the author?

---

Probably not, however in my opinion a function should error on failure instead of silently doing nothing.

---

Right well I was implying you could do a `SF.Throw` with a useful error if that happened - not no-op 👍

---

Ah, that makes far more sense, will do